### DISCUSSION TITLE
Mongodb thread 'tokio-runtime-worker' panicked at 'called `Option::unwrap()`

### DISCUSSION THREAD

**Discussion by masterbater** [prisma/prisma#12851] [Q&A]
```
thread 'tokio-runtime-worker' panicked at 'called `Option::unwrap()` on a `None` value', query-engine/connectors/mongodb-query-connector/src/root_queries/write.rs:49:52
stack backtrace:
   0: rust_begin_unwind
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/panicking.rs:498:5
   1: core::panicking::panic_fmt
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/panicking.rs:116:14
   2: core::panicking::panic
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/panicking.rs:48:5
   3: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
   4: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
   5: query_core::interpreter::query_interpreters::write::execute::{{closure}}
   6: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
   7: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
   8: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
   9: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  10: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  11: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  12: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  13: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  14: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  15: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  16: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  17: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  18: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  19: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  20: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  21: tokio::runtime::task::harness::poll_future
  22: tokio::runtime::task::harness::Harness<T,S>::poll
  23: std::thread::local::LocalKey<T>::with
  24: tokio::runtime::thread_pool::worker::Context::run_task
  25: tokio::runtime::thread_pool::worker::Context::run
  26: tokio::macros::scoped_tls::ScopedKey<T>::set
  27: tokio::runtime::thread_pool::worker::run
  28: <tokio::runtime::blocking::task::BlockingTask<T> as core::future::future::Future>::poll
  29: tokio::runtime::task::harness::Harness<T,S>::poll
  30: tokio::runtime::blocking::pool::Inner::run
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

**Comment by masterbater** [comment]
Prisma catches everything from typos, wrong fields, required fields but when we encountered this error  Option::unwrap() on a None value its hard to check what went wrong

I saw a lot of errors like this in issues and this is kinda hard to debug quickly, I believe the issue might be in my existing collection and the types generated by npx prisma db pull introspection

**Reply by janpio** [reply]
Indeed, we are crashing pretty brutally on the _unexpected_ things so we get the maximum amount of information. Adding nicer error handling here would be similar to fixing the underlying problem to get the relevant information, and then crash anyway out of spite - not useful.

Can you maybe pinpoint it to which model this is causing? And maybe even the data we would need to reproduce this?

**Comment by masterbater** [comment]
Solving the underlying inconsistent data types fix all the problems with
the error. Introspection only scan around 1000 records so some
inconsistencies are being skip. Prisma studio helps me confirm this since
it did spit some errors and guides me to fix those. Prisma is powerful
enough and very happy 😊 working with it.

**Reply by janpio** [reply]
Awesome.

Can you let us know which inconsistencies caused these problems? In theory this should not crash in this bad way but with a more helpful error message.

**Reply by janpio** [reply]
A colleague let me know that the stack trace you posted above indicates this happened while _writing_ data - which means that inconsistent data should not really be the reason as that mostly affects _reading_. Does that make sense?

**Comment by masterbater** [comment]
Yes it was during writing data logs with relation and the relation has some
inconsistent type data but Im not sure if that is the cause of error. I did
fix those and the panic error is gone. I cant provide replication for now
so I posted here in discussion instead of bug report.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
